online-rest-api/src/test/java/eu/itesla_project/online/rest/api/test/ProcessApiTest.java

```diff
@@ -0,0 +1,718 @@
+            return null;
+        }
+
+        @Override
+        public Map<Integer, List<LimitViolation>> getPostContingencyViolations(String workflowId,
+                String contingencyId) {
+            return null;
+        }
+
+        @Override
+        public Map<Integer, Map<String, List<LimitViolation>>> getPostContingencyViolations(String workflowId) {
+            if (!"1122".equals(workflowId))
+                return null;
+            Map<Integer, Map<String, List<LimitViolation>>> res = new HashMap<>();
+            Map<String, List<LimitViolation>> mmap = new HashMap<>();
+            List<LimitViolation> viols = new ArrayList<>();
+            LimitViolation lv = new LimitViolation(network.getIdentifiable("sub1"), LimitViolationType.CURRENT, 100,
+                    "HIGH_CURRENT", 0, 200, Country.FR, 120);
+            viols.add(lv);
+            mmap.put("test_contingency", viols);
+            res.put(0, mmap);
+            return res;
+        }
+
+        @Override
+        public Map<String, Boolean> getPostContingencyLoadflowConvergence(String workflowId, Integer stateId) {
+            return null;
+        }
+
+        @Override
+        public Map<Integer, Boolean> getPostContingencyLoadflowConvergence(String workflowId, String contingencyId) {
+            return null;
+        }
+
+        @Override
+        public Map<Integer, Map<String, Boolean>> getPostContingencyLoadflowConvergence(String workflowId) {
+            if (!"1122".equals(workflowId))
+                return null;
+            Map<Integer, Map<String, Boolean>> res = new HashMap<>();
+            Map<String, Boolean> mmap = new HashMap<>();
+            mmap.put("test_contingency", Boolean.TRUE);
+            res.put(0, mmap);
+            return res;
+        }
+
+        @Override
+        public void storeProcess(OnlineProcess proc) throws Exception {
+        }
+
+        @Override
+        public List<OnlineProcess> listProcesses() throws IOException {
+            return processMap.values().stream().collect(Collectors.toList());
+        }
+
+        @Override
+        public OnlineProcess getProcess(String processId) throws IOException {
+            if("error".equals(processId))
+                throw new IOException();
+            return processMap.get(processId);
+        }
+
+        @Override
+        public List<String[]> getAllMetrics(String workflowId, OnlineStep step) {
+            return null;
+        }
+
+        @Override
+        public void storeState(String workflowId, Integer stateId, Network network, String contingencyId) {
+            
+        }
+
+        @Override
+        public Map<Integer, Set<String>> listStoredPostContingencyStates(String workflowId) {
+            return null;
+        }
+
+        @Override
+        public Network getState(String workflowId, Integer stateId, String contingencyId) {
+            return null;
+        }
+
+    }
+
+}
```

Comment:
The mock classes could be extracted to independant class files to limit the file size and make it easier to read